she/evolve.py

```
@@ -0,0 +1,328 @@
+    if incident.state in _TERMINAL_BLOCK:
+        raise EvolveError(
+            f"evolution not applicable for terminal state {incident.state.value}"
+        )
+    if _is_security(incident):
+        raise EvolveError("security/dependabot incidents are observe-only")
+
+    sandbox = sandbox or plan_repair_sandbox(incident)
+    verification = verification or plan_verification(incident, sandbox=sandbox)
+    _bind_children(incident, sandbox, verification)
+    repair = plan_repair_pr(incident, sandbox=sandbox, verification=verification)
+    gates = tuple(sorted(set(verification.required_gates()) | set(repair.required_tests) | DUAL_GATES))
```

Pylint (4.0.7)

[convention] Line too long (103/100)

(C0301)